### PR TITLE
Require `moderate_comments` capability to context=edit a Comment

### PR DESCRIPTION
**Pull request by danielbachhuber**
If a comment author created a comment without the `moderate_comments`
capability, use context=view

Don't use `edit_comment` capability because `moderate_comments` can be
used to view the Comment list table without granting edit access

See #950, #702

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@WP-API/amigos #reviewmerge

File: tests/test-json-comments-controller.php [hunk 1]
```diff
@@ -116,6 +117,13 @@ public function test_get_comment_invalid_id() {
 		$this->assertErrorResponse( 'json_comment_invalid_id', $response, 404 );
 	}
 
+	public function test_get_comment_invalid_context() {
+		$request = new WP_JSON_Request( 'GET', sprintf( '/wp/comments/%s', $this->approved_id ) );
+		$request->set_param( 'context', 'edit' );
+		$response = $this->server->dispatch( $request );
+		$this->assertErrorResponse( 'json_comment_cannot_view', $response, 403 );
```
Comment (joehoyle) [inline tests/test-json-comments-controller.php]: With the latest merge of permissions checks, I'd expect this to return a generic `json_forbidden` now instead.

File: lib/endpoints/class-wp-json-comments-controller.php [hunk 2]
```diff
@@ -47,6 +47,10 @@ public function get_item( $request ) {
 			return new WP_Error( 'json_comment_invalid_id', __( 'Invalid comment ID.' ), array( 'status' => 404 ) );
 		}
 
+		if ( ! empty( $request['context'] ) && 'edit' === $request['context'] && ! current_user_can( 'moderate_comments' ) ) {
```
Comment (joehoyle) [inline lib/endpoints/class-wp-json-comments-controller.php]: We should move this check to the permissions check callback instead me things now

File: lib/endpoints/class-wp-json-comments-controller.php [hunk 3]
```diff
@@ -329,6 +337,10 @@ public function get_item_permissions_check( $request ) {
 			return false;
 		}
 
+		if ( ! empty( $request['context'] ) && 'edit' === $request['context'] && ! current_user_can( 'moderate_comments' ) ) {
+			return new WP_Error( 'json_forbidden', __( 'Sorry, you cannot view this comment with edit context' ), array( 'status' => 403 ) );
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-comments-controller.php]: Shouldn't this return `bool`?

Comment (joehoyle) [inline lib/endpoints/class-wp-json-comments-controller.php]: The caps check can return either. I kept the `WP_Error` so we can have a custom error message in this case.

Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-comments-controller.php]: Can you update the PHPdoc then?

Comment (joehoyle) [inline lib/endpoints/class-wp-json-comments-controller.php]: 804cadd

**Pull request merged**